### PR TITLE
1506 program requisitions display program data in requisition

### LINKED ISSUE #1506: Program Requisitions - Display program data in requisition
From epic: #1451 

Blocker: #1504 

Once api is exposed in above, we should be able to do al of the UI changes for requisition list and detail view:

* Requisition List (response and request), add these columns (if program requisitions are configured)
   * Program name
   * Order type
   * Period name
* When displaying request requisition should
   * Show program
   * Show order type
   * Show period
   * ~~Disable changes to supplier, maxMOS, thresholdMOS, period, order type, program, item addition/deletion~~ in another issue ? https://github.com/openmsupply/open-msupply/issues/1507
 * When displaying response requisition should
   * Show program
   * Show order type
   * Show period
   * ~~Disable addition/deletion of items~~ in another issue ? https://github.com/openmsupply/open-msupply/issues/1507

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #1506 

# 👩🏻‍💻 What does this PR do? 
 

image

image

# 🧪 How has/should this change been tested? 
Haven't done a lot of testing.
I created the program information manually in my database as I was waiting on #1481 

## 💌 Any notes for the reviewer?

Currently this doesn't remove the columns if programs aren't in use.
This could be addressed in a later issue.

## 📃 Documentation

- [ ] Program information visible on Requisitions (ResponseRequistions) and Internal Orders (RequestRequistitions)

### CODE REVIEW

**Comment by mark-prins** [comment]
Can I check the intention of the program table? 
The id field has a FK to the `master_list.id` and there's also a `master_list_id` column which also has an FK on `master_list.id`. Can we remove one of these?
Also, the data that has come in from mSupply has an mSupply sourced program id and because that isn't a master_list id I can't use it. I would revert back to the mSupply usage of an `id` distinct from the `master_list.id` - would you prefer to drop the `master_list_id` column and update the translation? 

Keeping them distinct means that we can change the master list which is associated with a program. Is that valid though?

File: client/packages/requisitions/src/ResponseRequisition/DetailView/Toolbar.tsx [hunk 1]
```diff
@@ -70,11 +80,53 @@ export const Toolbar: FC = () => {
...
+                />
               )}
             </Box>
           </Box>
+          <Box display="flex" flexDirection="row" gap={4}>
+            <Box display="flex" flex={1} flexDirection="column" gap={1}></Box>
+            <Box display="flex" flex={4} flexDirection="column" gap={6}>
+              {showInfo && <InfoPanel message={t('info.no-shipment')} />}
+            </Box>
+          </Box>
```
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/DetailView/Toolbar.tsx:129]: The panel is squashed up - and I don't see the need. How does this look?
Also, no need for the whole box bit if the `showInfo` is false, so I lifted that up
```suggestion
          {showInfo && (
            <Box padding={2}>
              <InfoPanel message={t('info.no-shipment')} />
            </Box>
          )}
```

was:
Screenshot 2023-04-20 at 8 59 11 AM


now:

Screenshot 2023-04-20 at 9 02 11 AM

File: client/packages/requisitions/src/ResponseRequisition/ListView/ListView.tsx [hunk 2]
```diff
@@ -62,6 +62,32 @@ export const ResponseRequisitionListView: FC = () => {
...
         width: 50,
       },
+      {
+        key: 'programName',
+        accessor: ({ rowData }) => {
+          return rowData.programName;
+        },
+        label: 'label.program',
+        description: 'description.program',
+        sortable: true,
```
Comment (mark-prins) [inline client/packages/requisitions/src/ResponseRequisition/ListView/ListView.tsx:72]: can you remove sortable from these too please

**Review by mark-prins — CHANGES_REQUESTED** [review]
Thanks! looking good. I have a slightly related concern about data structure, which I now think has been made on the wrong PR, soz. 
Some other requested changes in the comments

File: client/packages/requisitions/src/RequestRequisition/ListView/ListView.tsx [hunk 3]
```diff
@@ -58,6 +58,33 @@ export const RequestRequisitionListView: FC = () => {
...
+        sortable: true,
+      },
+
+      {
+        key: 'period',
+        accessor: ({ rowData }) => {
+          return rowData.period?.name ?? '';
+        },
+        label: 'label.period',
+        sortable: true,
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply client/packages/requisitions/src/RequestRequisition/ListView/ListView.tsx:85]: Thought I'd just add sorting support instead while I'm here...

**Review by mark-prins — APPROVED** [review]
Looks great thanks!

**Pull request merged**